rework baotha boring lesser heal

Antoonij opened the pull request:
## About The Pull Request
Baotha lesser heal:

Non-humans (carbons, simplemobs) receive a generic message and no bonus.

If the target is human, the ability checks for certain status effects:
Having druqks (Baotha's Blessing) or drunk buff grants +0.5 bonus.
Having stress event from Last High spell +0.5 bonus.

Then it checks all wounds of the target and sums their woundpain + bleed_rate (raw_suffering), with the following exceptions:
If the target has TRAIT_NOPAIN without TRAIT_CRACKHEAD, the suffering bonus is not applied.
If the target has TRAIT_NOPAIN and TRAIT_CRACKHEAD, the suffering bonus is applied.
Targets without TRAIT_NOPAIN receive the suffering bonus normally.
(Baotha's followers always have Crackhead trait. To heal other nopainers, just cast Baotha's Blessing on them or give them the Rosa Ring.)

A suffering-based bonus is calculated as sqrt(raw_suffering) / 3.535(roughly 1 fracture or some slashes/hematomas with pain_mod = 1 grants us full bonus).
That suffering bonus is capped at 2 (since woundpain reduces very quickly, full bonus isn't consistent) and is applied as follows:
If the target has TRAIT_DEPRAVED (Baotha follower), they receive the full calculated suffering bonus (ignores pain_mod).
Other humans have the suffering bonus scaled by physiology.pain_mod.

The maximum situational bonus is capped at 3.

## Testing Evidence
<img width="1755" height="909" alt="image" src="https://github.com/user-attachments/assets/b6c054b0-e2e3-4b06-8668-7ceb0999333f" />


## Why It's Good For The Game
smol rework to the boring thing.
no longer need to hold breath to get the bonus....

squidqueen commented:
Necra's also becomes significantly harder, corpses in view of 5 becomes the single hardest heal in the game tbh

Antoonij commented:
> balancejakk summary this is actually a massive nerf to both outside of necrans spamming self heal around undead. Anyone who has 2 fractures or massively stacked wounds would have hit the 50% health threshold for baothans but very well may not reach the pain to even get full use of it, and penalizes painkillers despite baothan missionaries having 3 mechanical ways to help with reducing pain
> 
> Necrans losing the ability to stop ppl dying in their miracle is also bad, and quite crippling. Scaling from corpses or graves if it was swapped would still suck. Healing if undead are onscreen is also The Most Situational Lesser Miracle in the game. Also given that anyone with lesser heal is likely either a support or a paladin, they shouldn't have garbage heals when applied to others
> 
> Potential solutions if change is actually desired by balancejakk maintainers: if status effect: high = increased heal for baothan miracle. It fits the lore and their vibe, hedonistic heal etc
> 
> further specialize necrans into preventing death and that's it with their miracle = make it restore a ton more blood if blood is low

Reverted Necra's lesser miracle. Lowered Baotha's pain divider (max effect now at 100 pain).  
Added bypass for TRAIT_NOPAIN if the target has TRAIT_CRACKHEAD — allows self-healing with nopain and healing other nopain targets under Baotha's blessing.

squidqueen commented:
> Reverted Necra's lesser miracle. Lowered Baotha's pain divider (max effect now at 100 pain). Added bypass for TRAIT_NOPAIN check if the target has TRAIT_CRACKHEAD — allows self-healing with nopain and healing other nopain targets under Baotha's blessing.

Healing based off of current wounds is also how the psydonite 'lesser miracle' endure works, though functionally a little differently because it's not pain but total damage + total wounds. 

The change is probably better at 100pain but still works inverse to Baotha's toolkit. The baotha's blessing power halves pain, herozium removes pain, and joyride applies no_pain straight up. I still think a weaker maybe 2.0 heal but applied on the High status effect would be the most consistent. But I can live with this if it makes it thru at least

bearrrrrrrr commented:
...wouldn't this actually make it so that healing yourself on ozium is bad, or am i misreading this and stupid?
